includes/compat/subscriptions/trait-wc-payment-gateway-wcpay-subscriptions.php

```diff
@@ -131,6 +140,19 @@ public function maybe_init_subscriptions() {
 
 		$this->supports = array_merge( $this->supports, $payment_gateway_features );
 
+		/**
+		 * The following callbacks are only attached once to avoid duplication.
+		 * The callbacks are also only intended to be attached for the WCPay core payment gateway ($this->id = 'woocommerce_payments').
+		 *
+		 * If new payment method IDs (eg 'sepa_debit') are added to this condition in the future, care should be taken to ensure duplication,
+		 * including double renewal charging, isn't introduced.
+		 */
+		if ( self::$has_attached_integration_hooks || 'woocommerce_payments' !== $this->id ) {
```

Comment:
The part about the previous solutions that I did not like was that it would have prevented using SEPA for subscriptions, as it would be just the one gateway, which gets subscriptions initialized.

Now I realize that without split-UPE, there is just one gateway. And a couple of seconds ago, I saw that having SEPA enabled makes the account ineligible for split-UPE anyway.

So practically both solutions have the same effect right now.